File: GoLang/quotes/linearService.go
package main

import (
	"context"
	"time"

	pb "github.com/RYANCOAL9999/AI_Go_Proto_File"
	"github.com/gin-gonic/gin"
	"google.golang.org/grpc"
	"google.golang.org/grpc/credentials/insecure"
)

type linearGRPCClient struct {
	client pb.LinearServiceClient
}

func newLinearGRPCClient(addr string) (*linearGRPCClient, error) {
	conn, err := grpc.Dial(addr, grpc.WithTransportCredentials(insecure.NewCredentials()))
	if err != nil {
		return nil, err
	}
	return &linearGRPCClient{
		client: pb.NewLinearServiceClient(conn),
	}, nil
}

func linearRegressionResponse(linearClient *linearGRPCClient) gin.HandlerFunc {
	return func(c *gin.Context) {

		const event_name = "linear regression"

		var data pb.LinearRegressionRequest
		if err := readRequestData(c, &data); err != nil {
			handleBadRequestError(c, err, event_name)
			return
		}

		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
		defer cancel()

		response, err := linearClient.client.LinearRegressionEvent(ctx, &data)

		if err != nil {
			handleGRPCError(c, err, event_name)
			return
		}

		sendResponse(c, response, event_name)

	}
}

func ridgeResponse(linearClient *linearGRPCClient) gin.HandlerFunc {
	return func(c *gin.Context) {

		const event_name = "ridge"

		var data pb.LinearRidgeRequest
		if err := readRequestData(c, &data); err != nil {
			handleBadRequestError(c, err, event_name)
			return
		}

		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
		defer cancel()

		response, err := linearClient.client.LinearRidgeEvent(ctx, &data)

		if err != nil {
			handleGRPCError(c, err, event_name)
			return
		}

		sendResponse(c, response, event_name)
	}

}

func ridgeCVResponse(linearClient *linearGRPCClient) gin.HandlerFunc {
	return func(c *gin.Context) {

		const event_name = "ridge cv"

		var data pb.LinearRidgeCVRequest
		if err := readRequestData(c, &data); err != nil {
			handleBadRequestError(c, err, event_name)
			return
		}

		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
		defer cancel()

		response, err := linearClient.client.LinearRidgeCVEvent(ctx, &data)

		if err != nil {
			handleGRPCError(c, err, event_name)
			return
		}

		sendResponse(c, response, event_name)

	}

}

func lassoResponse(linearClient *linearGRPCClient) gin.HandlerFunc {
	return func(c *gin.Context) {

		const event_name = "lasso"

		var data pb.LassoExpressionRequest
		if err := readRequestData(c, &data); err != nil {
			handleBadRequestError(c, err, event_name)
			return
		}

		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
		defer cancel()

		response, err := linearClient.client.LassoExpressionEvent(ctx, &data)

		if err != nil {
			handleGRPCError(c, err, event_name)
			return
		}

		sendResponse(c, response, event_name)
	}

}

func lassoLarsResponse(linearClient *linearGRPCClient) gin.HandlerFunc {
	return func(c *gin.Context) {

		const event_name = "lasso lars"

		var data pb.LassoLarsLassoExpressionRequest
		if err := readRequestData(c, &data); err != nil {
			handleBadRequestError(c, err, event_name)
			return
		}

		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
		defer cancel()

		response, err := linearClient.client.LassoLarsLassoExpressionEvent(ctx, &data)

		if err != nil {
			handleGRPCError(c, err, event_name)
			return
		}

		sendResponse(c, response, event_name)

	}

}

func bayesianRidgeResponse(linearClient *linearGRPCClient) gin.HandlerFunc {
	return func(c *gin.Context) {

		const event_name = "bayesian ridge"

		var data pb.BayesianRidgeRequest
		if err := readRequestData(c, &data); err != nil {
			handleBadRequestError(c, err, event_name)
			return
		}

		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
		defer cancel()

		response, err := linearClient.client.BayesianRidgeEvent(ctx, &data)

		if err != nil {
			handleGRPCError(c, err, event_name)
			return
		}

		sendResponse(c, response, event_name)

	}

}

func tweedieRegressorResponse(linearClient *linearGRPCClient) gin.HandlerFunc {
	return func(c *gin.Context) {

		const event_name = "tweedie regressor"

		var data pb.TweedieRegressorRequest
		if err := readRequestData(c, &data); err != nil {
			handleBadRequestError(c, err, event_name)
			return
		}

		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
		defer cancel()

		response, err := linearClient.client.TweedieRegressorEvent(ctx, &data)

		if err != nil {
			handleGRPCError(c, err, event_name)
			return
		}

		sendResponse(c, response, event_name)

	}

}

func sgdClassifierResponse(linearClient *linearGRPCClient) gin.HandlerFunc {
	return func(c *gin.Context) {

		const event_name = "sgd classifier"

		var data pb.SGDClassifierRequest
		if err := readRequestData(c, &data); err != nil {
			handleBadRequestError(c, err, event_name)
			return
		}

		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
		defer cancel()

		response, err := linearClient.client.SGDClassifierEvent(ctx, &data)

		if err != nil {
			handleGRPCError(c, err, event_name)
			return
		}

		sendResponse(c, response, event_name)

	}

}

func elasticNetResponse(linearClient *linearGRPCClient) gin.HandlerFunc {
	return func(c *gin.Context) {

		const event_name = "elastic net"

		var data pb.ElasticNetRequest
		if err := readRequestData(c, &data); err != nil {
			handleBadRequestError(c, err, event_name)
			return
		}

		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
		defer cancel()

		response, err := linearClient.client.ElasticNetEvent(ctx, &data)

		if err != nil {
			handleGRPCError(c, err, event_name)
			return
		}

		sendResponse(c, response, event_name)

	}

}

File: Python/quotes/model/linearService.py
import grpc

from dataclasses import asdict
from lib.proto.py.linear_expression_pb2_grpc import LinearServiceStub
from lib.proto.py.general_expression_pb2 import (
    LinearRegressionRequest,
    LinearRidgeRequest,
    LinearRidgeCVRequest,
    LassoExpressionRequest,
    LassoLarsLassoExpressionRequest,
    BayesianRidgeRequest,
    TweedieRegressorRequest,
    SGDClassifierRequest,
    ElasticNetRequest
) 

class LinearService(object):
    def __init__(
            self, 
            url:str
        )->None:
        self.__channel = grpc.insecure_channel(url)
        self.__stub = LinearServiceStub(self.__channel)

    def linearRegressionResponse(
            self, 
            data
        )->dict:
        with self.__channel:
            request_message = LinearRegressionRequest(
                x_drop_data = data.x_drop_data,
                y_drop_data = data.y_drop_data,
                size = data.size,
                random = data.random,
                key = data.key,
                sample_weight = data.sample_weight | None,
                kwargs = data.kwargs
            )
            response = self.__stub.LinearRegressionEvent(request_message)
            return asdict(response)
        
    def ridgeResponse(
            self, 
            data
        )->dict:
        with self.__channel:
            request_message = LinearRidgeRequest(
                x_drop_data = data.x_drop_data,
                y_drop_data = data.y_drop_data,
                size = data.size,
                random = data.random,
                key = data.key,
                sample_weight = data.sample_weight | None,
                kwargs = data.kwargs
            )
            response = self.__stub.LinearRidgeEvent(request_message)
            return asdict(response)
        
    def ridgeCVResponse(
            self, 
            data
        )->dict:
        with self.__channel:
            request_message = LinearRidgeCVRequest(
                x_drop_data = data.x_drop_data,
                y_drop_data = data.y_drop_data,
                size = data.size,
                random = data.random,
                key = data.key,
                sample_weight = data.sample_weight | None,
                kwargs = data.kwargs
            )
            response = self.__stub.LinearRidgeCVEvent(request_message)
            return asdict(response)
        
    def lassoResponse(
            self, 
            data
        )->dict:
        with self.__channel:
            request_message = LassoExpressionRequest(
                alpha = data.alpha,
                x_drop_data = data.x_drop_data,
                y_drop_data = data.y_drop_data,
                size = data.size,
                random = data.random,
                key = data.key,
                sample_weight = data.sample_weight | None,
                kwargs = data.kwargs
            )
            response = self.__stub.LassoExpressionEvent(request_message)
            return asdict(response)
        
    def lassoLarsResponse(
            self, 
            data
        )->dict:
        with self.__channel:
            request_message = LassoLarsLassoExpressionRequest(
                alpha = data.alpha,
                x_drop_data = data.x_drop_data,
                y_drop_data = data.y_drop_data,
                size = data.size,
                random = data.random,
                key = data.key,
                sample_weight = data.sample_weight | None,
                kwargs = data.kwargs
            )
            response = self.__stub.LassoLarsLassoExpressionEvent(request_message)
            return asdict(response)
        
    def BayesianRidgeResponse(
            self, 
            data
        )->dict:
        with self.__channel:
            request_message = BayesianRidgeRequest(
                x_drop_data = data.x_drop_data,
                y_drop_data = data.y_drop_data,
                size = data.size,
                random = data.random,
                key = data.key,
                sample_weight = data.sample_weight | None,
                kwargs = data.kwargs
            )
            response = self.__stub.BayesianRidgeEvent(request_message)
            return asdict(response)
        
    def tweedieRegressorResponse(
            self, 
            data
        )->dict:
        with self.__channel:
            request_message = TweedieRegressorRequest(
                alpha = data.alpha,
                x_drop_data = data.x_drop_data,
                y_drop_data = data.y_drop_data,
                size = data.size,
                random = data.random,
                key = data.key,
                sample_weight = data.sample_weight | None,
                kwargs = data.kwargs
            )
            response = self.__stub.TweedieRegressorEvent(request_message)
            return asdict(response)
        
    def sgdClassifierResponse(
            self, 
            data
        )->dict:
        with self.__channel:
            request_message = SGDClassifierRequest(
                x_drop_data = data.x_drop_data,
                y_drop_data = data.y_drop_data,
                size = data.size,
                random = data.random,
                key = data.key,
                sample_weight = data.sample_weight | None,
                kwargs = data.kwargs
            )
            response = self.__stub.SGDClassifierEvent(request_message)
            return asdict(response)
        
    def elasticNetResponse(
            self, 
            data
        )->dict:
        with self.__channel:
            request_message = ElasticNetRequest(
                alpha = data.alpha,
                x_drop_data = data.x_drop_data,
                y_drop_data = data.y_drop_data,
                size = data.size,
                random = data.random,
                key = data.key,
                sample_weight = data.sample_weight | None,
                kwargs = data.kwargs
            )
            response = self.__stub.ElasticNetEvent(request_message)
            return asdict(response)

